features/netsocket/UDPSocket.cpp

```diff
@@ -33,6 +33,17 @@ nsapi_protocol_t UDPSocket::get_proto()
     return NSAPI_UDP;
 }
 
+int UDPSocket::join_multicast_group(const SocketAddress &address)
+{
+    nsapi_ip_mreq_t mreq;
+
+    // Set up group address
+    mreq.imr_multiaddr = address.get_addr();
+    mreq.imr_interface = SocketAddress().get_addr();   // 0.0.0.0 INADDR_ANY
+
+    return this->setsockopt(0, NSAPI_ADD_MEMBERSHIP, &mreq, sizeof(mreq));
```

Comment:
This calls into Socket's setsockopt function, first param is level:
[See here](https://github.com/ARMmbed/mbed-os/blob/master/features/netsocket/Socket.cpp#L121)